interface/ceed-preconditioning.c

```diff
@@ -800,99 +664,46 @@ static int CeedSingleOperatorAssemble(CeedOperator op, CeedInt offset,
   ierr = CeedElemRestrictionDestroy(&rstr_q); CeedChk(ierr);
 
   // we store B_mat_in, B_mat_out, BTD, elem_mat in row-major order
-  CeedScalar B_mat_in[(num_qpts * num_eval_mode_in) * elem_size];
-  CeedScalar B_mat_out[(num_qpts * num_eval_mode_out) * elem_size];
```

Comment:
I'd imagine that caching these matrices rather than rebuilding them for every single element, each time we assemble the operator, will speed up CPU assembly.